### PR TITLE
Python 3: Port to Python3 for 60 py files

### PR DESCRIPTION
**Pull request by czhang0**
currently, keep to use commands module until we get final agreement about coding type and usage of process module of avocado.
Signed-off-by: cuzhang <cuzhang@redhat.com>

### CODE REVIEW

**Comment by czhang0** [comment]
Part of test result：
# avocado run --vt-type libvirt local..normal_test.show_schedinfo.valid_domname
JOB ID     : a8b22ea201a195ffa7c7b8b31729507f300a5bc1
JOB LOG    : /root/avocado/job-results/job-2018-03-09T14.36-a8b22ea/job.log
 (1/2) type_specific.local.virsh.schedinfo_xen_credit.normal_test.show_schedinfo.valid_domname: PASS (18.76 s)
 (2/2) type_specific.local.virsh.schedinfo_qemu_posix.normal_test.show_schedinfo.valid_domname: PASS (2.15 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 23.14 s
avocado run --vt-type libvirt local..normal_test.set_cpu_param.set_cpu_shares.value_exceed.set_by_cmd.config.valid_domid
JOB ID     : 3d5879e50cc59ff69e1198d9db5e0b8018862315
JOB LOG    : /root/avocado/job-results/job-2018-03-09T14.35-3d5879e/job.log
 (1/1) type_specific.local.virsh.schedinfo_qemu_posix.normal_test.set_cpu_param.set_cpu_shares.value_exceed.set_by_cmd.config.valid_domid: PASS (19.31 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 21.43 s

File: libvirt/tests/src/virsh_cmd/domain/virsh_reset.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def run(test, params, env):
     vm = env.get_vm(vm_name)
     domid = vm.get_id()
     domuuid = vm.get_uuid()
-    bef_pid = commands.getoutput("pidof -s qemu-kvm")
+    bef_pid = process.system_output("pidof -s qemu-kvm", shell=True)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_reset.py]: use junxiang's newly introduced getoutput method in process.py

File: libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml.py [hunk 2]
```diff
@@ -110,8 +111,8 @@ def pci_devices_address():
     if os.path.exists(polkit_file):
         replace_cmd = "sed -i 's/'ENTER.YOUR.PCI.DEVICE'/%s/g' /etc/polkit-1/rules.d/500-libvirt-acl-virttest.rules" % device_name
         cat_cmd = "cat /etc/polkit-1/rules.d/500-libvirt-acl-virttest.rules"
-        replace_output = commands.getoutput(replace_cmd)
-        cat_output = commands.getoutput(cat_cmd)
+        replace_output = process.system_output(replace_cmd, shell=True)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/nodedev/virsh_nodedev_dumpxml.py]: see previous comment

File: libvirt/tests/src/virsh_cmd/pool/virsh_pool.py [hunk 3]
```diff
@@ -346,7 +346,7 @@ def check_pool_info(pool_info, check_point, value):
         try:
             pvt.cleanup_pool(pool_name, pool_type, pool_target,
                              emulated_image, **kwargs)
-        except error.TestFail, detail:
+        except error.TestFail as detail:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/pool/virsh_pool.py]: please remove dependence on error module

File: libvirt/tests/src/virsh_cmd/secret/virsh_secret_define_undefine.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import os
-import commands
 
 from autotest.client import utils
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/secret/virsh_secret_define_undefine.py]: remove this dependency

File: libvirt/tests/src/virsh_cmd/secret/virsh_secret_define_undefine.py [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
+from avocado.utils import process
 from autotest.client.shared import error
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/secret/virsh_secret_define_undefine.py]: remove this dependency

File: libvirt/tests/src/virsh_cmd/secret/virsh_secret_dumpxml.py [hunk 6]
```diff
@@ -1,10 +1,11 @@
 import os
 import re
-import commands
+
 import tempfile
 
 from autotest.client.shared import error
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/secret/virsh_secret_dumpxml.py]: remove this dependency

File: libvirt/tests/src/virsh_cmd/secret/virsh_secret_list.py [hunk 7]
```diff
@@ -1,9 +1,9 @@
 import os
 import re
-import commands
 
 from autotest.client.shared import error
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/secret/virsh_secret_list.py]: remove this dependence

File: libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot.py [hunk 8]
```diff
@@ -167,7 +167,7 @@ def check_info(i1, i2, errorstr="Values differ"):
             test_file(session, sni["to_delete"], 2)
         except process.CmdError:
             handle_error("Failed to revert snapshot", vm_name)
-        except (error.TestFail, virt_vm.VMDeadError), e:
+        except (error.TestFail, virt_vm.VMDeadError) as e:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot.py]: remover error dependency

File: libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_create_as.py [hunk 9]
```diff
@@ -1,13 +1,13 @@
 import re
 import os
 import time
-import commands
 import string
 import logging
 
 from autotest.client.shared import error
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_create_as.py]: remove this dependence

File: libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_dumpxml.py [hunk 10]
```diff
@@ -126,7 +126,7 @@ def run(test, params, env):
             try:
                 vm_xml.VMXML.add_security_info(
                     vm_xml.VMXML.new_from_dumpxml(vm_name), passwd)
-            except Exception, info:
+            except Exception as info:
                 raise error.TestNAError(info)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_dumpxml.py]: remove this dependence

File: libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_edit.py [hunk 11]
```diff
@@ -57,7 +57,7 @@ def edit_snap_xml(dom_name, edit_opts, edit_cmd):
             remote.handle_prompts(session, None, None, r"[\#\$]\s*$")
             session.close()
             logging.info("Succeed to do snapshot edit")
-        except (aexpect.ShellError, aexpect.ExpectError), details:
+        except (aexpect.ShellError, aexpect.ExpectError) as details:
             log = session.get_output()
             session.close()
             raise error.TestFail("Failed to do snapshot-edit: %s\n%s"
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/snapshot/virsh_snapshot_edit.py]: remove error module depency

**Comment by czhang0** [comment]
Depend on avocado-framework/avocado#2510

**Pull request merged**